### PR TITLE
Data model

### PR DESCRIPTION
**Pull request by mpa-mxiang**
For this PR, I:

- Create a new app with a meaningful name.
- Initialize your project with Git.
- Has Postgres database set up.
- Build your project schema.
- Create and run the necessary migration files.
- Table and column names should match the ERD diagram.
- note: photo for the users table should be a link to an image
- Foreign keys should be included.
- Pay attention that in the posts table the column holding the foreign key to the users table should be named author_id
- All columns that are foreign keys should have a corresponding index.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,132 @@
...
+
+I would like to Microverse community for helping me to improve my skills.
+
+<p align="right"><a href="#readme-top">Back to top</a></p>
+
+<!-- FAQ (optional) -->
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](./LICENSE) licensed.
```
Comment (DammyShittu) [inline README.md:129]: - [x] Kindly fix your MIT file link. It is broken.

File: db/schema.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+#
+# It's strongly recommended that you check this file into your version control system.
+
+ActiveRecord::Schema[7.0].define(version: 2023_08_21_150440) do
+  # These are extensions that must be enabled in order to support this database
+  enable_extension "plpgsql"
+
+  create_table "comments", force: :cascade do |t|
+    t.bigint "userId_id", null: false
+    t.bigint "postId_id", null: false
```
Comment (DammyShittu) [inline db/schema.rb]: - [x] From the diagram in the requirements guideline, these should be `author_id` and `post_id`. Kindly rename them.

File: db/schema.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    t.string "text"
```
Comment (DammyShittu) [inline db/schema.rb]: - [x] The column type for the comment text column should be `text`, not `string`. Kindly fix this.

File: db/schema.rb [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.index ["postId_id"], name: "index_comments_on_postId_id"
+    t.index ["userId_id"], name: "index_comments_on_userId_id"
+  end
+
+  create_table "likes", force: :cascade do |t|
+    t.bigint "userId_id", null: false
+    t.bigint "postId_id", null: false
```
Comment (DammyShittu) [inline db/schema.rb]: - [x] From the diagram in the requirements guideline, these should be `author_id` and `post_id`. Kindly rename them.

File: db/schema.rb [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    t.index ["postId_id"], name: "index_likes_on_postId_id"
+    t.index ["userId_id"], name: "index_likes_on_userId_id"
+  end
+
+  create_table "posts", force: :cascade do |t|
+    t.bigint "authorId_id", null: false
+    t.string "title"
+    t.string "text"
+    t.integer "commentsCounter"
+    t.integer "likesCounter"
```
Comment (DammyShittu) [inline db/schema.rb]: - [x] From the diagram in the requirements guideline, these should be `author_id`. Kindly rename it.

- [x] Kindly rename these columns to `comments_counter` and `likes_counter` based on naming conventions

File: db/schema.rb [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    t.index ["authorId_id"], name: "index_posts_on_authorId_id"
+  end
+
+  create_table "users", force: :cascade do |t|
+    t.string "name"
+    t.string "bio"
+    t.string "photo"
+    t.date "updatedAt"
+    t.date "createdAt"
+    t.integer "postsCounter"
```
Comment (DammyShittu) [inline db/schema.rb]: - [x] In the users table, kindly remove the `updatedAt` and `createdAt` columns. We already have timestamp columns auto-generated by rails.

- [x] Kindly rename from `postsCounter` to `posts_counter` per conventions.

File: db/schema.rb [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    t.integer "commentsCounter"
+    t.integer "likesCounter"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.index ["authorId_id"], name: "index_posts_on_authorId_id"
+  end
+
+  create_table "users", force: :cascade do |t|
+    t.string "name"
+    t.string "bio"
```
Comment (DammyShittu) [inline db/schema.rb]: - [x] The bio column should have `text` data type, not `string`. Kindly fix that.

File: db/schema.rb [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.index ["postId_id"], name: "index_likes_on_postId_id"
+    t.index ["userId_id"], name: "index_likes_on_userId_id"
+  end
+
+  create_table "posts", force: :cascade do |t|
+    t.bigint "authorId_id", null: false
+    t.string "title"
+    t.string "text"
```
Comment (DammyShittu) [inline db/schema.rb]: - [x] Kindly change the data type for text to "text" as contained in the diagram. The "text" data type has more storage capacity for data, compared to string.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @mpa-mxiang,

## Highlights
✔️ Postgres db config

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Munsa1 — APPROVED** [review]
Hi @mpa-mxiang, 
congrats 4

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Made All Requested Changes :heavy_check_mark: 
Cleaned Up Migrations :heavy_check_mark: 
Clean CodeBase :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mpa-mxiang** [comment]
Thank you @Munsa1  for the approval!

**Pull request merged**